include/gatekeeper_acl.h

```diff
@@ -67,16 +67,34 @@ ipv4_acl_enabled(struct gatekeeper_if *iface)
 	 * The IPv4 ACL is only needed for interfaces that have
 	 * IPv4 addresses and don't support the ntuple filter.
 	 */
-	return !iface->hw_filter_ntuple && ipv4_if_configured(iface);
+	return !hw_filter_ntuple_available(iface) && ipv4_if_configured(iface);
 }
 
 static inline int
 ipv6_acl_enabled(struct gatekeeper_if *iface)
 {
-	/* The IPv6 ACL is needed whenever an interface has an IPv6 address. */
+	/*
+	 * The IPv6 ACL is needed whenever an interface has an IPv6 address.
+	 *
+	 * XXX #466 When ntuple filters are supported for IPv6, this
+	 * expression should check hw_filter_ntuple_available().
+	 */
 	return ipv6_if_configured(iface);
 }
 
+static inline int
+acl_enabled(struct gatekeeper_if *iface) {
```

Comment:
1. Let's add `const` to the definition of `iface`. This will require adding const to the called functions as well.
2. Let's change the return type from `int` to `bool`. This is not cosmetic because the code is relying on the return being `bool`, not just different from zero when true. We should change the return types of the called functions to `bool` as well.